.teamcity/test/AllProjectsTest.kt

```diff
@@ -61,7 +60,8 @@ class AllProjectsTest {
             if (project.buildTypes.size > 0) {
                 assertTrue(buildTypesIDs.containsKey(project), "Cannot find build types IDs for project ${project.name}")
                 val ids = project.buildTypes.map { it.id.toString() }.toSet()
-                assertEquals(buildTypesIDs[project], ids, "Build types IDs for project ${project.name} doesn't match")
+                val mismatchIDs = buildTypesIDs[project]!!.subtract(ids)
```

Comment:
Can we avoid use of !!? Should we be able to replace it with proper check or error handling.

Reply:
`assertTrue` few lines above checks this already but the compiler is stupid enough to detect it. So we will get NPE only if the assert fails.